dhctl/pkg/operations/bootstrap/module_configs.go

```diff
@@ -0,0 +1,186 @@
+			return fmt.Errorf("create ModuleConfig: %w", err)
+		}
+
+		modulesEnabledStatuses := computeModuleEnabledStatuses(initConfiguration.ConfigOverrides)
+		for moduleName, moduleEnabled := range modulesEnabledStatuses {
+			configOverride := initConfiguration.ConfigOverrides[moduleName]
+			settings := map[string]any{}
+			if configOverride != nil {
+				var settingsIsDict bool
+				settings, settingsIsDict = configOverride.(map[string]any)
+				if !settingsIsDict {
+					return fmt.Errorf("invalid configOverride, expected a dictionary, got %T", configOverride)
+				}
+			}
+
+			mc := buildUnstructuredModuleConfigWithOverrides(moduleName, moduleEnabled, settings)
+
+			if err := createModuleConfig(kubeCl.Dynamic(), mc); err != nil {
+				return fmt.Errorf("create ModuleConfig: %w", err)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("InitConfiguration conversion failed: %w", err)
+	}
+
+	err = log.Process("bootstrap", "Unlock bootstrap process", func() error {
+		if err := unlockBootstrapProcess(kubeCl); err != nil {
+			return fmt.Errorf("unlockBootstrapProcess: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// computeModuleEnabledStatuses loops over a InitConfiguration.deckhouse.configOverrides structure,
+// figuring out which ModuleConfig's should be enabled or disabled.
+// Returns a mapping of module name and a if it should be enabled or not.
+func computeModuleEnabledStatuses(configOverrides map[string]any) map[string]bool {
+	modulesEnabledStatuses := make(map[string]bool)
+	for key, override := range configOverrides {
+		moduleName, overrideIsEnableProperty := stringsutil.CutSuffix(key, "Enabled")
+		if overrideIsEnableProperty {
+			enabled, isBool := override.(bool)
+			if isBool {
+				modulesEnabledStatuses[moduleName] = enabled
+			} else {
+				// Avoid enabling module if it's config is malformed
+				modulesEnabledStatuses[moduleName] = false
+			}
+			continue
+		}
+
+		if _, enableStatusAlreadyDefined := modulesEnabledStatuses[moduleName]; !enableStatusAlreadyDefined {
+			// Enabling module by default if it has no %moduleName%Enabled property, skipping otherwise
+			modulesEnabledStatuses[moduleName] = true
+		}
+	}
+
+	return modulesEnabledStatuses
+}
+
+// buildUnstructuredModuleConfigWithOverrides creates ModuleConfig object as unstructured.Unstructured.
+func buildUnstructuredModuleConfigWithOverrides(moduleName string, isEnabled bool, settings map[string]any) *unstructured.Unstructured {
+	moduleConfigName := strcase.ToKebab(moduleName)
+	mc := &unstructured.Unstructured{}
+	mc.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "deckhouse.io",
+		Version: "v1alpha1",
+		Kind:    "ModuleConfig",
+	})
+	mc.SetName(moduleConfigName)
+
+	// Errors are impossible here.
+	_ = unstructured.SetNestedField(mc.Object, isEnabled, "spec", "enabled")
+	_ = unstructured.SetNestedField(mc.Object, int64(1), "spec", "version")
```

Comment:
interesting.

We can have version > 1 for module and for config with version 2 and may not have setting.
We need think about it.